packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -201,47 +206,18 @@ export class FeatureDevController {
         }
     }
 
-    // TODO add type
-    private async processUserChatMessage(message: any) {
-        if (message.message === undefined) {
-            this.messenger.sendErrorMessage('chatMessage should be set', message.tabID, 0, undefined, undefined)
-            return
-        }
-
-        /**
-         * Don't attempt to process any chat messages when a workspace folder is not set.
-         * When the tab is first opened we will throw an error and lock the chat if the workspace
-         * folder is not found
-         */
-        const workspaceFolders = vscode.workspace.workspaceFolders
-        if (workspaceFolders === undefined || workspaceFolders.length === 0) {
-            return
-        }
-
-        let session
-        try {
-            getLogger().debug(`${featureName}: Processing message: ${message.message}`)
+    private processErrorChatMessage = (err: any, message: any, session: Session | undefined) => {
+        const skipTitle = true
+        const errorMessage = createUserFacingErrorMessage(
+            `${featureName} request failed: ${err.cause?.message ?? err.message}`
+        )
 
-            session = await this.sessionStorage.getSession(message.tabID)
+        let isSendErrorMessage = true
```

Comment:
Why do we need this `isSendErrorMessage` boolean ?

Reply:
There is one scenario in the switch that we don't wanna throw the error with retries, but a final message. Maybe I can come up with a better naming for this.